src/cpp/session/SessionConsoleProcess.cpp

```diff
@@ -251,6 +253,67 @@ void ConsoleProcess::commonInit()
                            module_context::activeSession().id());
 #endif
    }
+   
+   // if RStudio has been configured to use Python,
+   // place that copy of Python on the PATH
+   std::string reticulatePython = core::system::getenv("RETICULATE_PYTHON");
+   if (reticulatePython.empty())
+   {
+      Error error = r::exec::RFunction(".rs.inferReticulatePython")
+            .call(&reticulatePython);
+      
+      if (error)
+         LOG_ERROR(error);
+      
+      if (!reticulatePython.empty())
+      {
+         core::system::setenv(
+                  &*options_.environment,
+                  "RETICULATE_PYTHON",
+                  reticulatePython);
+      }
+   }
+   
+   // if we're using bash, then we'll use a custom rcfile to ensure our
+   // hooks get run when the shell is launcehd
+   if (getShellType() == TerminalShell::ShellType::PosixBash ||
+       getShellType() == TerminalShell::ShellType::GitBash)
+   {
+      // we want to simulate a login shell, but without explicitly starting
+      // bash as a login shell. we do this so that we can control the rcfile
+      // sourced by bash on startup. to accomplish this, we run bash as an
+      // interactive non-login shell, but do the same startup initialization
+      // via our rcfile that a login shell would've performed
+      core::FilePath bashRcPath =
+            session::options().rResourcesPath().completeChildPath("terminal/bash/.bashrc");
+      
+      options_.args.emplace_back("--rcfile");
+      options_.args.emplace_back(bashRcPath.getAbsolutePath());
+   }
+   
+   // if we're using zsh, we want to set a custom ZDOTDIR so that we can
+   // inject some of our own tools to be run for newly-launched terminals.
+   // this allows us to invoke our own hooks while still running the user's
+   // own startup files as appropriate
+   if (getShellType() == TerminalShell::ShellType::PosixZsh)
+   {
+      FilePath zDotDir =
+            session::options().rResourcesPath().completeChildPath("terminal/zsh");
+      
+      core::system::setenv(
+               &*options_.environment,
+               "ZDOTDIR",
+               zDotDir.getAbsolutePath());
+   }
+ 
```

Comment:
Also, can we document a way to make this work with a custom shell config? People sometimes use Custom so they can tweak how we launch bash, zsh, etc.

Reply:
I've added a TODO for myself to update the terminal support article in the Juliet-rose timeframe, and will include mention of this feature. I haven't updated the article for 1.4 at all yet: https://support.rstudio.com/hc/en-us/articles/115010737148-Using-the-RStudio-Terminal